src/generate.cpp

```diff
@@ -87,14 +87,27 @@ bool SolveSpaceUI::PruneGroups(hGroup hg) {
 }
 
 bool SolveSpaceUI::PruneRequests(hGroup hg) {
-    auto e = std::find_if(SK.entity.begin(), SK.entity.end(),
-                          [&](Entity &e) { return e.group == hg && !EntityExists(e.workplane); });
-    if(e != SK.entity.end()) {
-        (deleted.requests)++;
-        SK.entity.RemoveById(e->h);
-        return true;
+    const int requests = SK.request.n;
+    for(Entity &e : SK.entity) {
+        if(e.group != hg) {
+            continue;
+        }
+
+        if(!e.h.isFromRequest()) {
```

Comment:
@iscgar Why did you decide to `continue` instead of:

```
        ssassert(e->h.isFromRequest(), "Only explicitly created entities can be pruned");
```

like [the old code did](https://github.com/solvespace/solvespace/commit/86f20cc7e5061d14323fdc0d814980922425e152#diff-d5fcd20826c6010f2f22908bf70fa6dfe7f2b186d5e151cb7fb4511adf41811bL101)?

Reply:
@ruevs  You can't assert since not all entities are from requests. It's searching for ones that are from requests, so we don't want to assert if they are not.